src/Provider/MetadataInterface.php

```diff
@@ -13,27 +13,58 @@
 
 namespace Sonata\MediaBundle\Provider;
 
-/**
- * NEXT_MAJOR: Remove CoreBundle dependency.
- */
-interface MetadataInterface extends \Sonata\CoreBundle\Model\MetadataInterface
-{
-    public function getTitle(): string;
+use Sonata\BlockBundle\Meta\MetadataInterface as BlockMetadataInterface;
```

Comment:
This one can be simplified:

```php
if (interface_exists(CoreMetadataInterface::class)) {
   interface InternalMetadataInterface extends CoreMetadataInterface, BlockMetadataInterface {}
 else {
   interface InternalMetadataInterface extends BlockMetadataInterface {}
}

interface MetadataInterface extends InternalMetadataInterface {
// ...
}
```

Reply:
I suggested to bind this metadata to the block bundle one, but maybe it is not necessary as you pointed out.

Should we remove that extends?

Reply:
If we follow BC very strict, we could not even remove the `CoreMetadataInterface` extension :/

But I would risk that case. If the user checks somewhere against for a `CoreMetadataInterface`, they should also require the CoreBundle in their `composer.json` file.